enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,329 @@
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceType string
+
+const (
+	// The DeviceTypes that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceType = "Disk"
+	// Part represents a device-type of partion
+	Partition DeviceType = "Partition"
+)
+
+type DeviceInclusionSpec struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty no devices will be selected.
+	Devices []DeviceType `json:"devices"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperties []DeviceMechanicalProperty `json:"deviceMechanicalProperties"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupStatus struct {
+	// Conditions is a list of conditions and their status.
+	Conditions []operatorv1.OperatorCondition `json:"conditions,omitempty"`
+
+	// TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+	TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+
+	// observedGeneration is the last generation change the operator has dealt with
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+```
+### Workflow of LocalPV creation via LocalVolumeGroup CR
+- Once the `LocalVolumeGroup` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
+
+#### Note: There is a chance of race condition and duplicate creation of PVs 
```

Comment:
From a resolved conversation looks like there is a follow-up separately from this PR to avoid or deal with the race condition. Can you update the design doc to explain the follow-up? Without that detail, it looks like there is still an issue in this design.